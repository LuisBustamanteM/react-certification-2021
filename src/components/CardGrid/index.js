export { default } from './CardGrid.component';