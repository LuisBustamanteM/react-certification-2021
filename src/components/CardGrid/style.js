import styled from 'styled-components';

export const CardGrid = styled.div`
    padding: 20px;
    margin: 0px auto;
    justify-content: flex-start;
    display: flex;
    flex-flow: row wrap;
    box-sizing: border-box;
    width: 1500px;
    @media screen and (max-width: 1500px) {
        width:1135px;
    }
    @media screen and (max-width: 1200px) {
        width:770px;
    }
    @media screen and (max-width: 800px) {
        width:100%;
        -moz-box-pack: center;
        justify-content: center;
    }
`;

export const ErrorMessage = styled.div`
  text-align: center;
  font-size: 24px;
  color: #965108;
  width: 100%;
`;