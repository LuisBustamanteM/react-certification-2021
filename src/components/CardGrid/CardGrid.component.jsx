import React from 'react';
import VideoCard from '../VideoCard';
import {CardGrid, ErrorMessage } from './style';

const CardGridComponent = ({items = [], errorMessage}) => {

    return(
        <CardGrid>
            { items.length > 0 ? items.map( (item) => (
                    <VideoCard key={item.id.videoId ? item.id.videoId : item.id}
                               id={item.id.videoId ? item.id.videoId : item.id}
                               defaultImage={item.snippet.thumbnails.default.url}
                               mediumImage={item.snippet.thumbnails.medium.url}
                               highImage={item.snippet.thumbnails.high.url}
                               title={item.snippet.title}
                               description={item.snippet.description}/>
                ))
                : <ErrorMessage>{errorMessage}</ErrorMessage>
            }
        </CardGrid>
    )
}

export default CardGridComponent;