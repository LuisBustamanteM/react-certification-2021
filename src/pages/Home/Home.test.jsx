import React, {useEffect, useState} from 'react';
import ReactDOM from  'react-dom';

import HomePage from './index';
import CardGrid from "../../components/CardGrid";
import {fireEvent, render, getByRole} from "@testing-library/react";



const build = () => {
    const { container, debug} = render(<HomePage/>)

    return {
        container,
        debug,
        searchbar: () => getByRole(container, "textbox"),
        submitButton: () => getByRole(container, "button", {name:"SEARCH"}),
        cardGrid: () => getByRole(container, "list") // TODO: Change to Ul
    }
}

describe("Testing <HomePage/>", () => {

    it("Triggers useFetch hook", () => {
        const {searchbar} = build();

    })

    // Test textbox

    // Test cardGrid

    // it("Displays Search bar and returns query parameters", () => {
    //    const {searchbar, debug} = build();
    //     let defaultState = "Wizeline";
    //     debug(searchbar());
    //     fireEvent.focus(searchbar());
    //     //
    //     // fireEvent.change(searchbar(), {target: {value: defaultState}});
    //     // expect(searchbar()).toHaveValue(defaultState);
    //     //
    //     // fireEvent.keyPress(searchbar(),  {key: "Enter"})
    //
    //
    // });
    // Test Youtube API
    // it("Must return Youtube Videos when requesting the API", () => {
    //
    //     // console.log(mockData);
    //     // Check Javascript YouTube API documentation
    //     // Check which API Call you need to get a list of videos by search parameter
    //     // Create a mock with jest so you dont overuse fetch
    // })

    // Test search parameters
    // it("Must return api search parameters", () => {
    //
    // })

    // Test youtube API setup
    //

    // test("<CardGrid/> is filled with items",  async () => {
    //     let items = await fetchApi("http://localhost:8080/videos")
    //         .then(data => data.items);
    //
    //     const component = render (<CardGrid items={items}/>);
    //
    //     expect (component.container).toMatchSnapshot();
    // });

    // test("<CardGrid/> Returns error message when it no longer has items",  () => {
    //     const component = render (<CardGrid />);
    //     expect (component.container).toMatchSnapshot();
    // });

});