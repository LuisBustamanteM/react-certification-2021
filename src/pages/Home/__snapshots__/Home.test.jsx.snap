// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomePage/> Compares with a snapshot of <HomePage/> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="homepage"
      >
        <h1
          class="sc-iBPTik cWzyta"
        >
          Welcome To The Challenge!
        </h1>
        <ul
          class="sc-jSgvzq cErgQG"
        >
          <h3
            class="sc-gKseQn flKMzi"
          >
            No videos to display, Try Reloading.
          </h3>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="homepage"
    >
      <h1
        class="sc-iBPTik cWzyta"
      >
        Welcome To The Challenge!
      </h1>
      <ul
        class="sc-jSgvzq cErgQG"
      >
        <h3
          class="sc-gKseQn flKMzi"
        >
          No videos to display, Try Reloading.
        </h3>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
